Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,185 @@
+        if (!m_viewport)
+        {
+            AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to find the a rendering viewport. Debug rendering will be disabled.");
+            return;
+        }
+
+        const auto fontQueryInterface = AZ::Interface<AzFramework::FontQueryInterface>::Get();
+        if (!fontQueryInterface)
+        {
+            AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to find the FontQueryInterface. Debug rendering will be disabled.");
+            return;
+        }
+
+        m_fontDrawInterface = fontQueryInterface->GetDefaultFontDrawInterface();
+        if (!m_fontDrawInterface)
+        {
+            AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to find the FontDrawInterface. Debug rendering will be disabled.");
+            return;
+        }
+
+        m_drawParams.m_drawViewportId = m_viewport->GetId();
+        m_drawParams.m_scale = AZ::Vector2(m_fontScale);
+        m_drawParams.m_color = m_fontColor;
+
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void NetworkDebugPlayerIdComponent::OnDeactivate([[maybe_unused]] EntityIsMigrating entityIsMigrating)
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void NetworkDebugPlayerIdComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        if (!cl_debugPlayerIdRender)
+        {
+            return;
+        }
+
+        AZ::Vector3 renderWorldSpace = GetEntity()->GetTransform()->GetWorldTranslation() + m_translationOffset;
+
+        // Don't render others players' on-screen debug text if the player is behind the camera
+        if (!IsNetEntityRoleAutonomous())
+        {
+            AZ::Vector3 cameraForward = m_viewport->GetCameraTransform().GetBasisY();
+            AZ::Vector3 cameraToPlayer = renderWorldSpace - m_viewport->GetCameraTransform().GetTranslation();
+            if (cameraForward.Dot(cameraToPlayer) < 0.0f)
+            {
+                return;
+            }   
+        }
+
+        const AzFramework::WindowSize windowSize = m_viewport->GetViewportSize();
+        AzFramework::ScreenPoint renderScreenpoint = AzFramework::WorldToScreen(
+            renderWorldSpace, m_viewport->GetCameraViewMatrixAsMatrix3x4(), m_viewport->GetCameraProjectionMatrix(), AzFramework::ScreenSize(windowSize.m_width, windowSize.m_height));
+        m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Center;
+        m_drawParams.m_position = AZ::Vector3(aznumeric_cast<float>(renderScreenpoint.m_x), aznumeric_cast<float>(renderScreenpoint.m_y), 0.f);
+
+        AZStd::string playerIdText = AZStd::string::format("Player %i", GetPlayerId());
+        m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, playerIdText);
+
+        if (IsNetEntityRoleAutonomous())
+        {
+            const auto controller = static_cast<NetworkDebugPlayerIdComponentController*>(GetController());
```

Comment:
Isn't this garatuneed if the role is autonomous?

Reply:
It well ought to be guaranteed! Hence the Assert; there must be a problem with the code itself should it be null. I guess that's too cautious? Better to leave out the null check entirely?

Reply:
removed the check